rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -1530,6 +1579,123 @@ RMW_Connext_Subscriber::take_serialized(
   return rc;
 }
 
+rmw_ret_t
+RMW_Connext_Subscriber::set_cft_expression_parameters(
+  const char * filter_expression,
+  const rcutils_string_array_t * expression_parameters)
+{
+  RMW_CONNEXT_ASSERT(nullptr != filter_expression)
+  RMW_CONNEXT_ASSERT(!internal)
+
+  rmw_ret_t ret;
+  std::lock_guard<std::mutex> lock(this->cft_mutex);
+  const bool filter_expression_empty = (*filter_expression == '\0');
+  if (nullptr == this->dds_topic_cft) {
+    // allow to call set filter_expresson even if cft not exist
+    if (filter_expression_empty) {
+      RMW_CONNEXT_LOG_DEBUG("current subscriber has no content filter topic")
+      return RMW_RET_OK;
+    }
+  } else {
+    if (!filter_expression_empty) {
+      // set filter expre if cft exist
+      return rmw_connextdds_set_cft_filter_expression(
+        this->dds_topic_cft, filter_expression, expression_parameters);
+    }
+  }
+
+  // finalization to remove the old data reader
+  ret = finalize(true);
```

Comment:
Please add a comment with the parameter name, so it's easier to read the code and know what the literal value is used for (e.g. `true /* reset_cft */`).

Reply:
updated